lib/mix/lib/mix/scm/git.ex

```diff
@@ -155,16 +156,25 @@ defmodule Mix.SCM.Git do
     if branch = opts[:branch] do
       "origin/#{branch}"
     else
-      opts[:ref] || opts[:tag] || "origin/master"
+      opts[:ref] || opts[:tag]
     end
   end
 
-  defp get_rev_info do
-    destructure [origin, rev],
-      :os.cmd('git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD')
+  defp get_rev_info(opts) do
+    cmd = 'git --git-dir=.git config remote.origin.url && git --git-dir=.git rev-parse --verify --quiet HEAD'
```

Comment:
Could we make the second command simply:

    rev_in_opts = opts[:tag] || opts[:branch] || opts[:rev] || "HEAD"
    cmd = "git --git-dir=.git rev-parse --verify --quiet #{rev_in_opts}~0"

Or do we need the third command? I think we just need to get the proper rev once.

If we need a third command, notice that you need to pass `--git-dir=.git` to the command you added below. :)